### PR TITLE
Implement a power law for the initial BH mass

### PR DESCRIPTION
**Pull request by sbird**
Implements a power law initial mass function for the BH

### CODE REVIEW

File: gadget/params.c [hunk 1]
```diff
@@ -198,6 +198,8 @@ create_gadget_parameter_set()
     param_declare_double(ps, "BlackHoleAccretionFactor", OPTIONAL, 100, "BH accretion boosting factor relative to the rate from the Bondi accretion model.");
     param_declare_double(ps, "BlackHoleEddingtonFactor", OPTIONAL, 2.1, "Maximum Black hole accretion as a function of Eddington.");
     param_declare_double(ps, "SeedBlackHoleMass", OPTIONAL, 2e-5, "Mass of initial black hole seed in internal mass units. If this is too much smaller than the gas particle mass, BH will not accrete.");
+    param_declare_double(ps, "MaxSeedBlackHoleMass", OPTIONAL, -1, "Black hole seed masses are drawn from a power law. This is the upper limit on the BH seed mass. If < 0 then all BHs have the SeedBlackHoleMass and the power law is disabled.");
```
Comment (rainwoodman) [inline gadget/params.c:201]: What about = 0?

Comment (sbird) [reply gadget/params.c:201]: Could work too

File: libgadget/blackhole.c [hunk 2]
```diff
@@ -1396,6 +1402,23 @@ blackhole_feedback_reduce(int place, TreeWalkResultBHFeedback * remote, enum Tre
     TREEWALK_REDUCE(BHP(place).CountProgs, remote->BH_CountProgs);
 }
 
+/* Sample from a power law to get the initial BH mass*/
+double get_bh_pwlaw_seed_mass(MyIDType ID)
```
Comment (rainwoodman) [inline libgadget/blackhole.c:1406]: powerlaw?

File: libgadget/blackhole.c [hunk 3]
```diff
@@ -1396,6 +1402,24 @@ blackhole_feedback_reduce(int place, TreeWalkResultBHFeedback * remote, enum Tre
     TREEWALK_REDUCE(BHP(place).CountProgs, remote->BH_CountProgs);
 }
 
+/* Sample from a power law to get the initial BH mass*/
+static double
+bh_powerlaw_seed_mass(MyIDType ID)
+{
+    /* compute random number, uniform in [0,1] */
+    const double w = get_random_number(ID);
```
Comment (rainwoodman) [inline libgadget/blackhole.c:1410]: Perhaps add an offset to ID to decorrelate with other random numbers seeded on the particle.

**Pull request merged**